File: app/src/main/java/com/example/dota2herodhevantharezamaap/java/MainActivity.java
package com.example.dota2herodhevantharezamaap.java;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;

import com.example.dota2herodhevantharezamaap.R;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    private RecyclerView rvHeroes;
    private ArrayList<Hero> list = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        rvHeroes = findViewById(R.id.rv_heroes);
        rvHeroes.setHasFixedSize(true);

        list.addAll(HeroesData.getListData());
        showRecyclerView();
    }
    protected void showRecyclerView() {
        rvHeroes.setLayoutManager(new LinearLayoutManager(this));
        HeroAdapter listHeroAdapter = new HeroAdapter(list);
        rvHeroes.setAdapter(listHeroAdapter);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.manu_main, menu);
        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        switch(item.getItemId()){
            case R.id.toProfile:
                Intent intent = new Intent(MainActivity.this, ProfilActivity.class);
                startActivity(intent);
                break;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/example/dota2herodhevantharezamaap/kotlin/activity/MainActivity.kt
package com.example.dota2herodhevantharezamaap.kotlin.activity

import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.dota2herodhevantharezamaap.R
import com.example.dota2herodhevantharezamaap.kotlin.HeroesData
import com.example.dota2herodhevantharezamaap.kotlin.adapter.HeroAdapter
import kotlinx.android.synthetic.main.activity_main.*
import org.jetbrains.anko.intentFor

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        showRecylerview()
    }

    private fun showRecylerview() {
        val listHero = HeroesData.getListData()

        rv_heroes.setHasFixedSize(true)
        rv_heroes.layoutManager = LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false)
        rv_heroes.adapter = HeroAdapter(this, listHero)
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.manu_main, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.toProfile -> startActivity(intentFor<ProfilActivity>())
        }
        return super.onOptionsItemSelected(item)
    }
}
